internal/benchrunner/runners/rally/runner.go

```diff
@@ -0,0 +1,816 @@
+		} `json:"hits"`
+	}
+
+	// Iterate through the search results using the Scroll API
+	for {
+		var sr searchRes
+		if err := json.NewDecoder(res.Body).Decode(&sr); err != nil {
+			return fmt.Errorf("error decoding search response: %w", err)
+		}
+
+		if sr.Error != nil {
+			return fmt.Errorf("error searching for documents: %s", sr.Error.Reason)
+		}
+
+		if len(sr.Hits) == 0 {
+			break
+		}
+
+		var bulkBodyBuilder strings.Builder
+		for _, hit := range sr.Hits {
+			bulkBodyBuilder.WriteString(fmt.Sprintf("{\"index\":{\"_index\":\"%s\",\"_id\":\"%s\"}}\n", indexName, hit.ID))
+			enriched := r.enrichEventWithBenchmarkMetadata(hit.Source)
+			src, err := json.Marshal(enriched)
+			if err != nil {
+				return fmt.Errorf("error decoding _source: %w", err)
+			}
+			bulkBodyBuilder.WriteString(fmt.Sprintf("%s\n", string(src)))
+		}
+
+		logger.Debugf("bulk request of %d events...", len(sr.Hits))
+
+		bulkRes, err := r.options.ESMetricsAPI.Bulk(strings.NewReader(bulkBodyBuilder.String()))
+		if err != nil {
+			return fmt.Errorf("error performing the bulk index request: %w", err)
+		}
+		bulkRes.Body.Close()
+
+		if sr.ScrollID == "" {
+			return errors.New("error getting scroll ID")
+		}
+
+		res, err = r.options.ESAPI.Scroll(
+			r.options.ESAPI.Scroll.WithScrollID(sr.ScrollID),
+			r.options.ESAPI.Scroll.WithScroll(time.Minute),
+		)
+		if err != nil {
+			return fmt.Errorf("error executing scroll: %s", err)
+		}
+		res.Body.Close()
+	}
+
+	logger.Debug("reindexing operation finished")
+	return nil
+}
+
+type benchMeta struct {
+	Info struct {
+		Benchmark string `json:"benchmark"`
+		RunID     string `json:"run_id"`
+	} `json:"info"`
+	Parameters scenario `json:"parameter"`
+}
+
+func (r *runner) enrichEventWithBenchmarkMetadata(e map[string]interface{}) map[string]interface{} {
+	var m benchMeta
+	m.Info.Benchmark = r.options.BenchName
+	m.Info.RunID = r.ctxt.Test.RunID
+	m.Parameters = *r.scenario
+	e["benchmark_metadata"] = m
+	return e
+}
+
+func getTotalHits(esapi *elasticsearch.API, dataStream string) (int, error) {
+	resp, err := esapi.Count(
+		esapi.Count.WithIndex(dataStream),
+	)
+	if err != nil {
+		return 0, fmt.Errorf("could not search data stream: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var results struct {
+		Count int
+		Error *struct {
+			Type   string
+			Reason string
+		}
+		Status int
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&results); err != nil {
+		return 0, fmt.Errorf("could not decode search results response: %w", err)
+	}
+
+	numHits := results.Count
+	if results.Error != nil {
+		logger.Debugf("found %d hits in %s data stream: %s: %s Status=%d",
+			numHits, dataStream, results.Error.Type, results.Error.Reason, results.Status)
+	} else {
+		logger.Debugf("found %d hits in %s data stream", numHits, dataStream)
+	}
+
+	return numHits, nil
+}
+
+func waitUntilTrue(fn func() (bool, error), timeout time.Duration) (bool, error) {
+	timeoutTicker := time.NewTicker(timeout)
+	defer timeoutTicker.Stop()
```

Comment:
Nit. This one doesn't need to be a ticker.
```suggestion
	timeout := time.NewTimer(timeout)
	defer timeout.Stop()
```